llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -483,23 +482,19 @@ attributeToExecModeMetadata(const Attribute &Attr, Function &F) {
                                             MDNode::get(Ctx, ClusterMDArgs));
   }
 
-  if ((AttrKindStr == SYCL_REGISTER_ALLOC_MODE_ATTR ||
-       AttrKindStr == SYCL_GRF_SIZE_ATTR) &&
-      !llvm::esimd::isESIMD(F)) {
+  if (AttrKindStr == SYCL_REGISTER_ALLOC_MODE_ATTR ||
+      AttrKindStr == SYCL_GRF_SIZE_ATTR) {
```

Comment:
Previously, the translator was looking for `RegisterAllocMode` MD to be `0`,`1` or `2`. With the new extension enabled, the translator looks for the actual GRF register size as the operand, which the `sycl-grf-size` attr already has, so now we need to fixup the old `sycl-register-alloc-mode` attr  to have the register size instead of the enum value.